src/tensor_ops/max_to/max_to.cu

```diff
@@ -31,34 +45,88 @@ extern "C" __global__ void fill_with(float *buf, float value, const size_t numel
     buf[i] = value;
 }
 
-// Accepts pre-broadcasted strides for both input & output.
-// So both inp & out are expected to be broadcasted to the same size.
+// Sourced from https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2
+__device__ __forceinline__ unsigned int next_power_of_two(unsigned int v) {
+    v--;
+    v |= v >> 1;
+    v |= v >> 2;
+    v |= v >> 4;
+    v |= v >> 8;
+    v++;
+    return v;
+}
+
+// Efficiently computes the max of each chunk in "data" of size chunk_len, and
+// stores the maximums in out[i / chunk_len]
+__device__ void chunk_max(
+    const size_t numel,
+    const size_t chunk_len,
+    const float data,
+    float* out
+) {
+    __shared__ float buf[1024];
+    // assumes that threads where i >= numel have already exited
+    unsigned int i = blockIdx.x * blockDim.x + threadIdx.x;
+    unsigned int block_i = threadIdx.x;
+    buf[block_i] = data;
```

Comment:
Ahhh is this where the strides come into play? The `data` value here is from inp[get_strided_index(...)], so each thread storing data in this shared buffer will be controlled by the strides.

Reply:
Yes, `data` is an array of values passed in to `chunk_sum` and others through each thread's registers.

Reply:
Gotcha, very nice!